### DISCUSSION TITLE
Allow users to delete their accounts

### DISCUSSION THREAD

**Discussion by tsmyre** [ManifoldScholar/manifold#3527] [Ideas]
Currently, only users with Admin roles can delete Manifold accounts. Users who create accounts simply to engage with projects and other users, as well as users who are given elevated roles to access the backend to create/edit projects, have no means to delete their own accounts if they so wish. We should make that possible through the `Edit Profile` user dropdown from the main menu.

Some considerations that come immediately to mind around enabling such a feature are:

- What happens to annotations and highlights left by a user who deletes their account? Right now, if a user is deleted those engagements are entirely removed. Is it worth considering retaining public annotations/comments/highlights and strip them of attribution? Or making that an option.
- Some users will likely want to retain a record of both their public and private annotations. What shape could that potentially take? JSON seems likely, but that format is not particularly useful, broadly speaking. What about HTML or Markdown?
- For users who have created projects, do we want to encourage users who are going to delete their accounts to first promote another user to manage their soon-to-be abandoned projects or to alert Admins that those projects no longer have an active user managing them?

**Comment by zdavis** [comment]
This has been implemented — accounts can be deleted in Manifold now.
